chaindexing-postgres/src/repo.ts

```diff
@@ -55,6 +55,23 @@ export class PostgresRepo extends Repo<Pool, Conn> {
     // TODO: Implement serial streaming version
   }
 
+  async streamContractAddresses_2(conn: Conn, limit = 10, allowOffset = true) {
+    let currentPage = 0;
+    let offset = 0;
+
+    const getNext = async (): Promise<ContractAddress[] | null> => {
+      if (allowOffset) {
+        offset = limit * currentPage;
+        currentPage += 1;
+      }
+      return conn.query.chaindexingContractAddressesSchema.findMany({ limit, offset });
+    };
+
+    return {
+      next: async () => await getNext()
```

Comment:
I am unsure why there are getNext and next fns. It seems repetitive. Would it be better to discard `getNext` and have the fn inline like below's suggestion?

```suggestion
      next: async (): Promise<ContractAddress[] | null> => {
         if (allowOffset) {
           offset = limit * currentPage;
           currentPage += 1;
         }
         return conn.query.chaindexingContractAddressesSchema.findMany({ limit, offset });
      }
```